### DISCUSSION TITLE
I am creating a PATCH request with REST API to update a record, but data is not updating.

### DISCUSSION THREAD

**Discussion by collegefam** [supabase/supabase#4824] [Questions] [answered]
I am creating a PATCH request with REST API to update a record, but data is not updating. What is the correct way to do it?
I'm running the following function for it:

 function delete_record(snumber){
    console.log("delete function has been called");
    let response =  fetch ("https://myprojectname.supabase.co/rest/v1/tablename?serialnumber=eq."+snumber, {
            method: "PATCH",
            headers: {"Content-type": "application/json", "apikey":"myservicekey", "Prefer": "return=representation","Authorization": "myservicekey"},
            body: JSON.stringify('{ "is_deleted": "1" }'),
        });
}

I am running function this in browser, through a button on my web app labelled as "Mark as deleted". When clicked, I want the column "is_deleted" to be labelled as '1' for that specific row.

**Comment by awalias** [comment, marked as answer]
hey @collegefam thanks for the question, could you share with us the response

from your code it looks like you may be missing `Bearer ` from the Authoirization header

it should look something like this:

```bash
curl -X PATCH 'https://<project-ref>.supabase.co/rest/v1/countries?some_column=eq.someValue' \
-H "apikey: SUPABASE_KEY" \
-H "Authorization: Bearer SUPABASE_KEY" \
-H "Content-Type: application/json" \
-H "Prefer: return=representation" \
-d '{ "other_column": "otherValue" }'
```

also it looks like you're stringifying a string, you may want to remove the single quotes and stringify the javascript object instead like ` JSON.stringify({ "is_deleted": "1" })`

**Reply by collegefam** [reply]
thanks, your solution worked.
